### PR TITLE
Feature/ui namespace wizard complete

### PR DESCRIPTION
**Pull request by PGuimarais**
- Enables user to create namespace and apply settings / preferences in wizard
- Adds KeyValuePairs component 
- Adds Redux store for KeyValuePairs

Bamboo Build:
http://builds.cask.co/browse/CDAP-DRC4685

### CODE REVIEW

File: cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePairs.less [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  padding-right: 10px;
+  cursor: pointer;
+}
+
+.add-row-btn {
+  color: black;
+}
+.remove-row-btn {
+  color: red;
+}
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePairs.less:44]: Nest all these inside a parent container, these changes are applied globally for the same class name

File: cdap-ui/app/cdap/components/KeyValuePairs/index.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      pairs: [...nextProps.keyValues.pairs]
+    });
+  }
+  render() {
+    return (
+      <div>
+      {
+        this.state.pairs.map( (field, index) => {
+          return (
+            <Provider store={this.keyValueStore} key={shortid.generate()}>
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/KeyValuePairs/index.js]: don't use shortid.generate here. Should be defined wherever the state.pairs being constructed

**Review by elfenheart — CHANGES_REQUESTED** [review]

File: cdap-ui/app/cdap/services/WizardStores/AddNamespace/ActionCreator.js [hunk 3]
```diff
@@ -21,14 +21,58 @@ const PublishNamespace = () => {
...
+  if(state.mapping.hiveDatabaseName){
+    putParams["config"]["hive.database"] = state.mapping.hiveDatabaseName;
+  }
+
+  if(state.mapping.hdfsDirectory){
+    putParams["config"]["root.directory"] = state.mapping.hdfsDirectory;
+  }
+
+  if(state.security.keytabURI){
+    putParams["config"]["keytabURI"] = state.security.keytabURI;
```
Comment (elfenheart) [inline cdap-ui/app/cdap/services/WizardStores/AddNamespace/ActionCreator.js]: check the proper state name of this keytab state

**Review by elfenheart — CHANGES_REQUESTED** [review]

**Comment by elfenheart** [comment]
screen shot 2016-10-17 at 4 42 28 pm

File: cdap-ui/app/cdap/components/KeyValuePairs/index.js (hunk 2 above)
Comment (PGuimarais) [reply cdap-ui/app/cdap/components/KeyValuePairs/index.js]: moved the unique id to the redux store for each pair

File: cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePairs.less (hunk 1 above)
Comment (PGuimarais) [reply cdap-ui/app/cdap/components/KeyValuePairs/KeyValuePairs.less:44]: done

**Comment by elfenheart** [comment]
The error name Failed to features.Wizard.undefined.headerlabel still happening

**Comment by elfenheart** [comment]
the preference store don't get reset i believe.

To reproduce:
1. Create a namespace, add preference
2. Create another namespace, go to the preference step, and the previous values will still be there

**Comment by ajainarayanan** [comment]
👍 LGTM

**Pull request merged**